internal/generator/mcyaml.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2023 Seagate Technology LLC and/or its Affiliates
+
+package generator
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"gopkg.in/yaml.v3"
+	"k8s.io/klog/v2"
+)
+
+const (
+	exceptionRename = "rename"
+)
+
+type GeneralInformation struct {
+	Title       string `yaml:"title"`
+	Description string `yaml:"description"`
+}
+
+type NestedMeta struct {
+	Name   string       `yaml:"name"`   // The name of the meta object, such as 'hosts-groups'
+	Nested []NestedMeta `yaml:"nested"` // When one or more response objects are nested into the main response object, such as 'redundancy' is nested under 'system'
+}
+
+type OptionInformation struct {
+	Flag            string `yaml:"flag"`             // The command line parameter option
+	Type            string `yaml:"type"`             // The option type: string, enum, abs
+	Values          string `yaml:"values"`           // Optional. Allowed enumeration values
+	Required        string `yaml:"required"`         // Required is true or false
+	KeywordRequired string `yaml:"keyword-required"` // Bool. Indicates keyword should be used in path, or must not be used in path
+	Description     string `yaml:"description"`      // Optional. Command description
+}
+
+type CommandInformation struct {
+	Path    string              `yaml:"path"`    // The command path, such as '/show/system'
```

Comment:
Replaced 'Path' variable with 'Command'. I used 'path' because that is the OpenAPI term and that is the point of this generator, to generate an OpenAPI spec, but it doesn't change the logic. Using spaces would not be clear at all since the OpenAPI uses slashes. I don't want to bend the OpenAPI specification for RESTful APIs to fit the MC API command line implementation.